resource/graph_node.go

```diff
@@ -111,6 +133,24 @@ func (w *GraphNode) Resource() (Resource, error) {
 	return w.current, nil
 }
 
+// WithState contains a resource along with lifecycle state information.
+type WithState struct {
```

Comment:
what will this be used for?

Reply:
Right now just for tests, but will eventually be exported in the robot API

Reply:
agree with @benjirewis  [here](https://github.com/viamrobotics/rdk/pull/4075/files#r1643234587), feels like we could just expose GraphNode without going through the extra steps

Reply:
true, i guess `GraphNode` is already supposed to be "resource with state"